core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/function/Concat.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.batch.aggregator.function;
+
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.data.schema.Schema.Type;
+import java.util.Objects;
+
+/**
+ * Concatenates the values in the group with a comma
+ */
+public class Concat implements AggregateFunction<String, Concat> {
+
+  private final String fieldName;
+  private final Schema fieldSchema;
+  private String concatString;
+  private boolean firstString = true;
+
+  public Concat(String fieldName, Schema fieldSchema) {
+    this.fieldName = fieldName;
+    this.fieldSchema = fieldSchema;
+    Type inputType =
+        fieldSchema.isNullable() ? fieldSchema.getNonNullable().getType() : fieldSchema.getType();
+
+    if (!inputType.equals(Type.STRING)) {
+      throw new IllegalArgumentException(
+          String.format("Field '%s' is of unsupported non-string type '%s'. ",
+              fieldName, inputType));
+    }
+  }
+
+  @Override
+  public void initialize() {
+    this.concatString = "";
+  }
+
+  @Override
+  public void mergeValue(StructuredRecord record) {
+    if (record.get(fieldName) != null) {
+      String value = Objects
+          .requireNonNull(record.get(fieldName)).toString();
+      if (firstString) {
```

Comment:
@CuriousVini this condition is only for the first string in order to avoid starting the string with the leading comma ex: "test1" instead of ", test1". The first statement after the if condition is used to merge the rest of the strings ```concatString = String.format("%s, %s", concatString, value);``` - line 62

Reply:
Will this method get executed by all the tasks in the framework? If so this will be applied to all the splits

Reply:
Correct, and split are also joined the same way - with String format - in which case also this check prevents situations like ``test1 ,, test2``.